vnext/Microsoft.ReactNative/ReactCoreInjection.idl

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import "IReactContext.idl";
+import "ReactNativeHost.idl";
+import "IJSValueReader.idl";
+
+#include "DocString.h"
+
+namespace Microsoft.ReactNative
+{
+  [experimental]
+  [webhosthidden]
+  DOC_STRING(
+    "Used to inject platform specific implementations to create react-native targets targeting non-XAML platforms.")
+  interface IReactCoreInjection
+  {
```

Comment:
consider naming it something more specific to its functionality, e.g. BatchCompleteHandler, BatchCompleteListener?

Maybe this can be a delegate instead?